fetcher.go

```diff
@@ -303,6 +318,9 @@ func (p *pool) fetchResource(ctx context.Context, from string, resource string,
 	}
 
 	if resp.StatusCode != http.StatusOK {
+		// empty body so it can be re-used.
+		_, _ = io.Copy(io.Discard, resp.Body)
```

Comment:
this is is a great bug to fix! maybe we quickly promote just this change and see how much it fixes?